public/app/reducers/municipality.js

```diff
@@ -1,16 +1,35 @@
 import types from '../actions/types';
-import municipalities from '../../assets/data/ma-munis.json';
+
+import colors from '~/app/constants/colors';
+import municipalities from '~/assets/data/ma-munis.json';
 
 const defaultState = {
   cache: municipalities.features.reduce((cache, feature) =>
       Object.assign({}, cache, { [feature.properties.town.toLowerCase()]: feature }), {}),
+  searchable: municipalities.features.reduce((a,b) => a.concat(b.properties.town.toLowerCase()), []),
+  geojson: municipalities,
 };
 
 export default function municipality(state = defaultState, action) {
+  let newState = {};
 
   switch(action.type) {
-    default:
-      return state;
+    case types.MUNICIPALITY.FILL_POLY:
+      const newGeoJSON = { ...state.geojson };
+
+      newGeoJSON.features.some(feature => {
+        if (feature.properties.town.toLowerCase() === action.muni) {
+          return true && (feature.properties.fillColor = colors.BRAND.PRIMARY);
+        }
+      });
```

Comment:
It's better practice in redux to treat everything as immutable, so this `.some` situation is an antipattern. Using `.some` as a linear search is also not very efficient.